src/ucp/core/ucp_am.c

```diff
@@ -1222,15 +1236,16 @@ ucs_status_t ucp_am_rndv_process_rts(void *arg, void *data, size_t length,
     ucp_am_rndv_rts_hdr_t *rts = data;
     ucp_worker_h worker        = arg;
     uint16_t am_id             = rts->am.am_id;
+    ucp_recv_desc_t *desc      = NULL;
     ucp_am_entry_t *am_cb;
-    ucp_recv_desc_t *desc;
+    ucp_request_t *req;
     ucp_am_recv_param_t param;
     ucs_status_t status, desc_status;
     void *hdr;
 
     if (ucs_unlikely(!ucp_am_recv_check_id(worker, am_id))) {
-        /* TODO: send ATS */
-        return UCS_OK;
+        status = UCS_ERR_REJECTED;
```

Comment:
is the `status` variable set by not used in `out_send_ats`?
should we return `status`?

Reply:
it is used on line 1291: ```ucp_rndv_req_send_ats(req, NULL, rts->super.sreq.req_id, status);```

Reply:
maybe send UCS_ERR_INVALID_PARAM?